File: Win/ui/views/settings_view.py
"""
SettingsView.py — DAS CRM Windows
User Preferences and Application Configuration
Feature parity with Android SettingsScreen.tsx
"""

from PyQt6.QtWidgets import (
    QWidget, QVBoxLayout, QHBoxLayout, QLabel, QPushButton, QLineEdit,
    QScrollArea, QFrame, QComboBox, QCheckBox, QSlider, QSpinBox,
    QMessageBox, QDialog, QTabWidget, QListWidget, QListWidgetItem
)
from PyQt6.QtCore import Qt
from PyQt6.QtGui import QFont, QColor
from dataclasses import dataclass

# ─────────────────────────────────────────────────────────────────────────────────────
# DATA MODELS
# ─────────────────────────────────────────────────────────────────────────────────────

@dataclass
class UserPreferences:
    """User preferences and settings"""
    userId: str
    userName: str
    email: str
    theme: str  # DARK, LIGHT, AUTO
    language: str  # EN, ES, FR, DE, JP, HI
    timezone: str
    notificationsEnabled: bool
    soundEnabled: bool
    emailDigest: str  # DAILY, WEEKLY, MONTHLY, NEVER
    autoSyncEnabled: bool
    syncInterval: int  # minutes

@dataclass
class ApiConfiguration:
    """API configuration settings"""
    baseUrl: str
    apiKey: str
    timeout: int  # seconds
    retryAttempts: int
    enableOfflineMode: bool

FALLBACK_PREFERENCES = UserPreferences(
    userId="u1",
    userName="John Sales Manager",
    email="john@dascrm.com",
    theme="DARK",
    language="EN",
    timezone="Asia/Kolkata",
    notificationsEnabled=True,
    soundEnabled=True,
    emailDigest="WEEKLY",
    autoSyncEnabled=True,
    syncInterval=5
)

# ─────────────────────────────────────────────────────────────────────────────────────
# MAIN SETTINGS VIEW
# ─────────────────────────────────────────────────────────────────────────────────────

class SettingsView(QWidget):
    """User Preferences and Application Configuration"""

    def __init__(self, parent=None):
        super().__init__(parent)
        self.setStyleSheet("""
            QWidget { background-color: #090d16; }
            QLabel { color: #f8fafc; }
            QPushButton { color: white; font-weight: bold; border: none; border-radius: 6px; }
            QLineEdit, QComboBox { background-color: #0f172a; color: #ffffff; border: 1px solid #1e293b;
                       border-radius: 6px; padding: 8px; }
            QCheckBox { color: #f8fafc; }
        """)

        self.preferences = FALLBACK_PREFERENCES

        self._build_ui()

    def _build_ui(self):
        """Build settings UI"""
        layout = QVBoxLayout(self)
        layout.setContentsMargins(0, 0, 0, 0)

        scrollArea = QScrollArea()
        scrollArea.setWidgetResizable(True)
        scrollArea.setStyleSheet("QScrollArea { border: none; background-color: #090d16; }")

        scrollWidget = QWidget()
        scrollLayout = QVBoxLayout(scrollWidget)
        scrollLayout.setContentsMargins(16, 16, 16, 24)
        scrollLayout.setSpacing(12)

        # Title
        titleLabel = QLabel("⚙️ Settings & Preferences")
        titleLabel.setFont(QFont("Segoe UI", 14, QFont.Weight.Bold))
        titleLabel.setStyleSheet("color: #ffffff;")
        scrollLayout.addWidget(titleLabel)

        # Account Section
        accountCard = self._build_section("👤 Account Settings", [
            ("Full Name", self.preferences.userName, False),
            ("Email", self.preferences.email, False),
            ("User ID", self.preferences.userId, True),
        ])
        scrollLayout.addWidget(accountCard)

        # Appearance Section
        appearanceCard = self._build_appearance_section()
        scrollLayout.addWidget(appearanceCard)

        # Notifications Section
        notificationsCard = self._build_notifications_section()
        scrollLayout.addWidget(notificationsCard)

        # Sync Settings Section
        syncCard = self._build_sync_section()
        scrollLayout.addWidget(syncCard)

        # API Configuration Section
        apiCard = self._build_api_section()
        scrollLayout.addWidget(apiCard)

        scrollLayout.addStretch()

        # Action Buttons
        actionLayout = QHBoxLayout()
        actionLayout.setContentsMargins(0, 12, 0, 0)
        actionLayout.setSpacing(8)

        btnSave = QPushButton("💾 Save Changes")
        btnSave.setStyleSheet("background-color: #10b981; padding: 8px 16px;")
        btnSave.clicked.connect(lambda: QMessageBox.information(self, "✓ Saved", "Settings saved successfully"))
        actionLayout.addWidget(btnSave)

        btnReset = QPushButton("↻ Reset to Defaults")
        btnReset.setStyleSheet("background-color: #6366f1; padding: 8px 16px;")
        btnReset.clicked.connect(lambda: QMessageBox.question(self, "⚠️ Reset?", "Reset all settings to defaults?"))
        actionLayout.addWidget(btnReset)

        btnClearCache = QPushButton("🗑️ Clear Cache")
        btnClearCache.setStyleSheet("background-color: #f97316; padding: 8px 16px;")
        btnClearCache.clicked.connect(lambda: QMessageBox.information(self, "✓ Cache Cleared", "Application cache cleared"))
        actionLayout.addWidget(btnClearCache)

        actionLayout.addStretch()
        scrollLayout.addLayout(actionLayout)

        scrollArea.setWidget(scrollWidget)
        layout.addWidget(scrollArea)

    def _build_section(self, title: str, fields: list) -> QFrame:
        """Build a settings section card"""
        card = QFrame()
        card.setStyleSheet("""
            QFrame {
                background-color: #0f172a;
                border: 1px solid #1e293b;
                border-radius: 12px;
                padding: 12px;
            }
        """)

        layout = QVBoxLayout(card)
        layout.setContentsMargins(12, 12, 12, 12)
        layout.setSpacing(12)

        titleLabel = QLabel(title)
        titleLabel.setFont(QFont("Segoe UI", 11, QFont.Weight.Bold))
        titleLabel.setStyleSheet("color: #ffffff;")
        layout.addWidget(titleLabel)

        for fieldName, fieldValue, isReadOnly in fields:
            fieldLayout = QVBoxLayout()
            fieldLayout.setContentsMargins(0, 0, 0, 0)
            fieldLayout.setSpacing(4)

            nameLabel = QLabel(fieldName)
            nameLabel.setFont(QFont("Segoe UI", 9, QFont.Weight.Bold))
            nameLabel.setStyleSheet("color: #94a3b8;")
            fieldLayout.addWidget(nameLabel)

            valueInput = QLineEdit()
            valueInput.setText(str(fieldValue))
            valueInput.setReadOnly(isReadOnly)
            valueInput.setStyleSheet(f"""
                QLineEdit {{
                    background-color: {'#020617' if isReadOnly else '#0f172a'};
                    color: #cbd5e1;
                    border: 1px solid #334155;
                    border-radius: 6px;
                    padding: 6px;
                }}
            """)
            fieldLayout.addWidget(valueInput)

            layout.addLayout(fieldLayout)

        return card

    def _build_appearance_section(self) -> QFrame:
        """Build appearance settings section"""
        card = QFrame()
        card.setStyleSheet("""
            QFrame {
                background-color: #0f172a;
                border: 1px solid #1e293b;
                border-radius: 12px;
                padding: 12px;
            }
        """)

        layout = QVBoxLayout(card)
        layout.setContentsMargins(12, 12, 12, 12)
        layout.setSpacing(12)

        titleLabel = QLabel("🎨 Appearance")
        titleLabel.setFont(QFont("Segoe UI", 11, QFont.Weight.Bold))
        titleLabel.setStyleSheet("color: #ffffff;")
        layout.addWidget(titleLabel)

        # Theme
        themeLayout = QVBoxLayout()
        themeLabel = QLabel("Theme")
        themeLabel.setFont(QFont("Segoe UI", 9, QFont.Weight.Bold))
        themeLabel.setStyleSheet("color: #94a3b8;")
        themeLayout.addWidget(themeLabel)

        themeCombo = QComboBox()
        themeCombo.addItems(["Dark", "Light", "Auto"])
        themeCombo.setCurrentText("Dark")
        themeLayout.addWidget(themeCombo)

        layout.addLayout(themeLayout)

        # Language
        langLayout = QVBoxLayout()
        langLabel = QLabel("Language")
        langLabel.setFont(QFont("Segoe UI", 9, QFont.Weight.Bold))
        langLabel.setStyleSheet("color: #94a3b8;")
        langLayout.addWidget(langLabel)

        langCombo = QComboBox()
        langCombo.addItems(["English", "Español", "Français", "Deutsch", "日本語", "हिंदी"])
        langCombo.setCurrentText("English")
        langLayout.addWidget(langCombo)

        layout.addLayout(langLayout)

        # Timezone
        tzLayout = QVBoxLayout()
        tzLabel = QLabel("Timezone")
        tzLabel.setFont(QFont("Segoe UI", 9, QFont.Weight.Bold))
        tzLabel.setStyleSheet("color: #94a3b8;")
        tzLayout.addWidget(tzLabel)

        tzCombo = QComboBox()
        tzCombo.addItems(["UTC", "Asia/Kolkata", "America/New_York", "Europe/London", "Asia/Tokyo"])
        tzCombo.setCurrentText("Asia/Kolkata")
        tzLayout.addWidget(tzCombo)

        layout.addLayout(tzLayout)

        return card

    def _build_notifications_section(self) -> QFrame:
        """Build notifications settings section"""
        card = QFrame()
        card.setStyleSheet("""
            QFrame {
                background-color: #0f172a;
                border: 1px solid #1e293b;
                border-radius: 12px;
                padding: 12px;
            }
        """)

        layout = QVBoxLayout(card)
        layout.setContentsMargins(12, 12, 12, 12)
        layout.setSpacing(12)

        titleLabel = QLabel("🔔 Notifications")
        titleLabel.setFont(QFont("Segoe UI", 11, QFont.Weight.Bold))
        titleLabel.setStyleSheet("color: #ffffff;")
        layout.addWidget(titleLabel)

        # Enable notifications
        notifCheck = QCheckBox("Enable Notifications")
        notifCheck.setChecked(True)
        layout.addWidget(notifCheck)

        # Sound
        soundCheck = QCheckBox("Enable Sound")
        soundCheck.setChecked(True)
        layout.addWidget(soundCheck)

        # Email digest
        digestLayout = QVBoxLayout()
        digestLabel = QLabel("Email Digest Frequency")
        digestLabel.setFont(QFont("Segoe UI", 9, QFont.Weight.Bold))
        digestLabel.setStyleSheet("color: #94a3b8;")
        digestLayout.addWidget(digestLabel)

        digestCombo = QComboBox()
        digestCombo.addItems(["Daily", "Weekly", "Monthly", "Never"])
        digestCombo.setCurrentText("Weekly")
        digestLayout.addWidget(digestCombo)

        layout.addLayout(digestLayout)

        return card

    def _build_sync_section(self) -> QFrame:
        """Build sync settings section"""
        card = QFrame()
        card.setStyleSheet("""
            QFrame {
                background-color: #0f172a;
                border: 1px solid #1e293b;
                border-radius: 12px;
                padding: 12px;
            }
        """)

        layout = QVBoxLayout(card)
        layout.setContentsMargins(12, 12, 12, 12)
        layout.setSpacing(12)

        titleLabel = QLabel("🔄 Sync Settings")
        titleLabel.setFont(QFont("Segoe UI", 11, QFont.Weight.Bold))
        titleLabel.setStyleSheet("color: #ffffff;")
        layout.addWidget(titleLabel)

        # Auto sync
        syncCheck = QCheckBox("Auto Sync Enabled")
        syncCheck.setChecked(True)
        layout.addWidget(syncCheck)

        # Sync interval
        intervalLayout = QVBoxLayout()
        intervalLabel = QLabel("Sync Interval (minutes)")
        intervalLabel.setFont(QFont("Segoe UI", 9, QFont.Weight.Bold))
        intervalLabel.setStyleSheet("color: #94a3b8;")
        intervalLayout.addWidget(intervalLabel)

        intervalSpinBox = QSpinBox()
        intervalSpinBox.setMinimum(1)
        intervalSpinBox.setMaximum(60)
        intervalSpinBox.setValue(5)
        intervalSpinBox.setStyleSheet("QSpinBox { background-color: #0f172a; color: #ffffff; border: 1px solid #334155; border-radius: 6px; }")
        intervalLayout.addWidget(intervalSpinBox)

        layout.addLayout(intervalLayout)

        # Offline mode
        offlineCheck = QCheckBox("Enable Offline Mode")
        offlineCheck.setChecked(True)
        layout.addWidget(offlineCheck)

        return card

    def _build_api_section(self) -> QFrame:
        """Build API configuration section"""
        card = QFrame()
        card.setStyleSheet("""
            QFrame {
                background-color: #0f172a;
                border: 1px solid #1e293b;
                border-radius: 12px;
                padding: 12px;
            }
        """)

        layout = QVBoxLayout(card)
        layout.setContentsMargins(12, 12, 12, 12)
        layout.setSpacing(12)

        titleLabel = QLabel("🔗 API Configuration")
        titleLabel.setFont(QFont("Segoe UI", 11, QFont.Weight.Bold))
        titleLabel.setStyleSheet("color: #ffffff;")
        layout.addWidget(titleLabel)

        # Base URL
        urlLayout = QVBoxLayout()
        urlLabel = QLabel("API Base URL")
        urlLabel.setFont(QFont("Segoe UI", 9, QFont.Weight.Bold))
        urlLabel.setStyleSheet("color: #94a3b8;")
        urlLayout.addWidget(urlLabel)

        urlInput = QLineEdit()
        urlInput.setText("http://localhost:4000/api")
        urlLayout.addWidget(urlInput)

        layout.addLayout(urlLayout)

        # API Key (masked)
        keyLayout = QVBoxLayout()
        keyLabel = QLabel("API Key")
        keyLabel.setFont(QFont("Segoe UI", 9, QFont.Weight.Bold))
        keyLabel.setStyleSheet("color: #94a3b8;")
        keyLayout.addWidget(keyLabel)

        keyInput = QLineEdit()
        keyInput.setText("••••••••••••••••")
        keyInput.setEchoMode(QLineEdit.EchoMode.Password)
        keyLayout.addWidget(keyInput)

        layout.addLayout(keyLayout)

        # Timeout
        timeoutLayout = QVBoxLayout()
        timeoutLabel = QLabel("Request Timeout (seconds)")
        timeoutLabel.setFont(QFont("Segoe UI", 9, QFont.Weight.Bold))
        timeoutLabel.setStyleSheet("color: #94a3b8;")
        timeoutLayout.addWidget(timeoutLabel)

        timeoutSpinBox = QSpinBox()
        timeoutSpinBox.setMinimum(5)
        timeoutSpinBox.setMaximum(120)
        timeoutSpinBox.setValue(30)
        timeoutSpinBox.setStyleSheet("QSpinBox { background-color: #0f172a; color: #ffffff; border: 1px solid #334155; border-radius: 6px; }")
        timeoutLayout.addWidget(timeoutSpinBox)

        layout.addLayout(timeoutLayout)

        # Test Connection Button
        testBtn = QPushButton("🧪 Test Connection")
        testBtn.setStyleSheet("background-color: #3b82f6; padding: 6px 12px;")
        testBtn.clicked.connect(lambda: QMessageBox.information(self, "✓ Connection OK", "Successfully connected to API"))
        layout.addWidget(testBtn)

        return card

File: macOs/Sources/DASCRM/Views/Settings/SettingsView.swift
"""
SettingsView.swift — DAS CRM macOS
User Preferences and Application Configuration
Feature parity with Android SettingsScreen.tsx
"""

import SwiftUI

struct UserPreferences {
    var userName: String = "John Sales Manager"
    var email: String = "john@dascrm.com"
    var userId: String = "u1"
    var theme: String = "DARK"
    var language: String = "EN"
    var timezone: String = "Asia/Kolkata"
    var notificationsEnabled: Bool = true
    var soundEnabled: Bool = true
    var emailDigest: String = "WEEKLY"
    var autoSyncEnabled: Bool = true
    var syncInterval: Int = 5
    var baseURL: String = "http://localhost:4000/api"
    var requestTimeout: Int = 30
}

@MainActor
class SettingsViewModel: ObservableObject {
    @Published var preferences = UserPreferences()
    @Published var showSuccessMessage = false
    @Published var successMessage = ""
}

struct SettingsView: View {
    @StateObject private var viewModel = SettingsViewModel()
    @State private var showConnectionTest = false

    var body: some View {
        VStack(alignment: .leading, spacing: 0) {
            // Header
            VStack(alignment: .leading, spacing: 12) {
                Text("⚙️ Settings & Preferences")
                    .font(.system(size: 14, weight: .bold))
                    .foregroundColor(.white)
            }
            .padding(16)
            .background(Color(red: 0.06, green: 0.09, blue: 0.16))
            .border(Color(red: 0.12, green: 0.16, blue: 0.23), width: 1)

            // Settings Scroll
            ScrollView {
                VStack(alignment: .leading, spacing: 12) {
                    // Account Section
                    SettingsSectionView(title: "👤 Account Settings") {
                        SettingsRowView(label: "Full Name", value: $viewModel.preferences.userName, isReadOnly: false)
                        SettingsRowView(label: "Email", value: $viewModel.preferences.email, isReadOnly: false)
                        SettingsRowView(label: "User ID", value: $viewModel.preferences.userId, isReadOnly: true)
                    }

                    // Appearance Section
                    SettingsSectionView(title: "🎨 Appearance") {
                        VStack(alignment: .leading, spacing: 8) {
                            Text("Theme")
                                .font(.system(size: 10, weight: .bold))
                                .foregroundColor(Color(red: 0.58, green: 0.64, blue: 0.68))

                            Picker("Theme", selection: $viewModel.preferences.theme) {
                                Text("Dark").tag("DARK")
                                Text("Light").tag("LIGHT")
                                Text("Auto").tag("AUTO")
                            }
                            .pickerStyle(.segmented)
                        }

                        VStack(alignment: .leading, spacing: 8) {
                            Text("Language")
                                .font(.system(size: 10, weight: .bold))
                                .foregroundColor(Color(red: 0.58, green: 0.64, blue: 0.68))

                            Picker("Language", selection: $viewModel.preferences.language) {
                                Text("English").tag("EN")
                                Text("Español").tag("ES")
                                Text("Français").tag("FR")
                                Text("Deutsch").tag("DE")
                                Text("日本語").tag("JP")
                                Text("हिंदी").tag("HI")
                            }
                            .pickerStyle(.menu)
                        }

                        VStack(alignment: .leading, spacing: 8) {
                            Text("Timezone")
                                .font(.system(size: 10, weight: .bold))
                                .foregroundColor(Color(red: 0.58, green: 0.64, blue: 0.68))

                            Picker("Timezone", selection: $viewModel.preferences.timezone) {
                                Text("UTC").tag("UTC")
                                Text("Asia/Kolkata").tag("Asia/Kolkata")
                                Text("America/New_York").tag("America/New_York")
                                Text("Europe/London").tag("Europe/London")
                                Text("Asia/Tokyo").tag("Asia/Tokyo")
                            }
                            .pickerStyle(.menu)
                        }
                    }

                    // Notifications Section
                    SettingsSectionView(title: "🔔 Notifications") {
                        Toggle("Enable Notifications", isOn: $viewModel.preferences.notificationsEnabled)
                            .foregroundColor(.white)

                        Toggle("Enable Sound", isOn: $viewModel.preferences.soundEnabled)
                            .foregroundColor(.white)

                        VStack(alignment: .leading, spacing: 8) {
                            Text("Email Digest Frequency")
                                .font(.system(size: 10, weight: .bold))
                                .foregroundColor(Color(red: 0.58, green: 0.64, blue: 0.68))

                            Picker("Email Digest", selection: $viewModel.preferences.emailDigest) {
                                Text("Daily").tag("DAILY")
                                Text("Weekly").tag("WEEKLY")
                                Text("Monthly").tag("MONTHLY")
                                Text("Never").tag("NEVER")
                            }
                            .pickerStyle(.menu)
                        }
                    }

                    // Sync Settings Section
                    SettingsSectionView(title: "🔄 Sync Settings") {
                        Toggle("Auto Sync Enabled", isOn: $viewModel.preferences.autoSyncEnabled)
                            .foregroundColor(.white)

                        VStack(alignment: .leading, spacing: 8) {
                            HStack {
                                Text("Sync Interval (minutes)")
                                    .font(.system(size: 10, weight: .bold))
                                    .foregroundColor(Color(red: 0.58, green: 0.64, blue: 0.68))
                                Spacer()
                                Text("\(viewModel.preferences.syncInterval)")
                                    .font(.system(size: 10, weight: .bold))
                                    .foregroundColor(Color(red: 0.38, green: 0.65, blue: 0.98))
                            }

                            Slider(value: Double(viewModel.preferences.syncInterval), in: 1...60, step: 1)
                                .onChange(of: viewModel.preferences.syncInterval) { newValue in
                                    viewModel.preferences.syncInterval = Int(newValue)
                                }
                        }

                        Toggle("Enable Offline Mode", isOn: $viewModel.preferences.autoSyncEnabled)
                            .foregroundColor(.white)
                    }

                    // API Configuration Section
                    SettingsSectionView(title: "🔗 API Configuration") {
                        SettingsRowView(label: "API Base URL", value: $viewModel.preferences.baseURL, isReadOnly: false)

                        VStack(alignment: .leading, spacing: 8) {
                            HStack {
                                Text("Request Timeout (seconds)")
                                    .font(.system(size: 10, weight: .bold))
                                    .foregroundColor(Color(red: 0.58, green: 0.64, blue: 0.68))
                                Spacer()
                                Text("\(viewModel.preferences.requestTimeout)")
                                    .font(.system(size: 10, weight: .bold))
                                    .foregroundColor(Color(red: 0.38, green: 0.65, blue: 0.98))
                            }

                            Slider(value: Double(viewModel.preferences.requestTimeout), in: 5...120, step: 5)
                                .onChange(of: viewModel.preferences.requestTimeout) { newValue in
                                    viewModel.preferences.requestTimeout = Int(newValue)
                                }
                        }

                        Button(action: { showConnectionTest = true }) {
                            Text("🧪 Test Connection")
                                .font(.system(size: 10, weight: .bold))
                                .foregroundColor(.white)
                                .frame(maxWidth: .infinity)
                                .padding(8)
                                .background(Color(red: 0.24, green: 0.51, blue: 0.96))
                                .cornerRadius(6)
                        }
                        .buttonStyle(.plain)
                    }

                    Spacer()

                    // Action Buttons
                    HStack(spacing: 8) {
                        Button(action: {
                            viewModel.successMessage = "Settings saved successfully"
                            viewModel.showSuccessMessage = true
                        }) {
                            Text("💾 Save Changes")
                                .font(.system(size: 10, weight: .bold))
                                .foregroundColor(.white)
                                .frame(maxWidth: .infinity)
                                .padding(8)
                                .background(Color(red: 0.06, green: 0.68, blue: 0.50))
                                .cornerRadius(6)
                        }
                        .buttonStyle(.plain)

                        Button(action: {
                            viewModel.preferences = UserPreferences()
                            viewModel.successMessage = "Settings reset to defaults"
                            viewModel.showSuccessMessage = true
                        }) {
                            Text("↻ Reset to Defaults")
                                .font(.system(size: 10, weight: .bold))
                                .foregroundColor(.white)
                                .frame(maxWidth: .infinity)
                                .padding(8)
                                .background(Color(red: 0.49, green: 0.40, blue: 0.94))
                                .cornerRadius(6)
                        }
                        .buttonStyle(.plain)

                        Button(action: {
                            viewModel.successMessage = "Application cache cleared"
                            viewModel.showSuccessMessage = true
                        }) {
                            Text("🗑️ Clear Cache")
                                .font(.system(size: 10, weight: .bold))
                                .foregroundColor(.white)
                                .frame(maxWidth: .infinity)
                                .padding(8)
                                .background(Color(red: 0.98, green: 0.58, blue: 0.09))
                                .cornerRadius(6)
                        }
                        .buttonStyle(.plain)
                    }
                }
                .padding(16)
            }
            .background(Color(red: 0.06, green: 0.08, blue: 0.14))
        }
        .background(Color(red: 0.06, green: 0.08, blue: 0.14))
        .navigationTitle("Settings")
        .alert("✓ Success", isPresented: $viewModel.showSuccessMessage) {
            Button("OK") { }
        } message: {
            Text(viewModel.successMessage)
        }
        .alert("🧪 Connection Test", isPresented: $showConnectionTest) {
            Button("OK") { }
        } message: {
            Text("Successfully connected to API at \(viewModel.preferences.baseURL)")
        }
    }
}

struct SettingsSectionView<Content: View>: View {
    let title: String
    @ViewBuilder let content: () -> Content

    var body: some View {
        VStack(alignment: .leading, spacing: 12) {
            Text(title)
                .font(.system(size: 12, weight: .bold))
                .foregroundColor(.white)
                .padding(.bottom, 4)

            VStack(alignment: .leading, spacing: 12) {
                content()
            }
            .padding(12)
            .background(Color(red: 0.06, green: 0.09, blue: 0.16))
            .border(Color(red: 0.12, green: 0.16, blue: 0.23), width: 1)
            .cornerRadius(12)
        }
    }
}

struct SettingsRowView: View {
    let label: String
    @Binding var value: String
    let isReadOnly: Bool

    var body: some View {
        VStack(alignment: .leading, spacing: 6) {
            Text(label)
                .font(.system(size: 10, weight: .bold))
                .foregroundColor(Color(red: 0.58, green: 0.64, blue: 0.68))

            if isReadOnly {
                Text(value)
                    .font(.system(size: 10))
                    .foregroundColor(Color(red: 0.79, green: 0.84, blue: 0.88))
                    .padding(8)
                    .background(Color(red: 0.02, green: 0.06, blue: 0.12))
                    .border(Color(red: 0.12, green: 0.16, blue: 0.23), width: 1)
                    .cornerRadius(6)
            } else {
                TextField(label, text: $value)
                    .textFieldStyle(.roundedBorder)
                    .foregroundColor(.white)
            }
        }
    }
}

#Preview {
    NavigationView {
        SettingsView()
    }
}
